internal/convert/database_user.go

```diff
@@ -112,44 +111,29 @@ func BuildAtlasScopes(r []string) []atlasv2.UserScope {
 	return scopes
 }
 
-// BuildLegacyAtlasScopes converts the scopes inside the array of string in an array of mongodbatlas.Scope structs.
-// r contains resources in the format resourceName:resourceType.
-// currently required for dbusers update only :: remove after CLOUDP-176215 is closed.
-func BuildLegacyAtlasScopes(r []string) []atlas.Scope {
-	scopes := make([]atlas.Scope, len(r))
-	for i, scopeP := range r {
-		scope := strings.Split(scopeP, scopeSep)
-		resourceType := defaultResourceType
-		if len(scope) > 1 {
-			resourceType = scope[1]
-		}
+// GetAuthDB determines the authentication database based on the type of user.
+// LDAP, X509 and AWSIAM should all use $external.
+// SCRAM-SHA should use admin.
+func GetAuthDB(user *atlasv2.DatabaseUser) string {
```

Comment:
fyi this logic was previously handled in the v1 sdk @wtrocki so it's a special case we should probably document
https://github.com/mongodb/go-client-mongodb-atlas/blob/master/mongodbatlas/database_users.go#L226

Reply:
@matt-condon Maybe we can copy paste this code into new example?
https://github.com/mongodb/atlas-sdk-go/tree/main/examples

Would you have time to do it since you already have context and required backend state?

Reply:
It looks like copy paste of that would be enough for us to showcase how to sent requests.

Reply:
Sure will do

Reply:
Provided here: https://github.com/mongodb/atlas-sdk-go/pull/55